CIPs/0002.md

```diff
@@ -0,0 +1,93 @@
+We should support a multitude of text message providers to provide geographic diversity and avoid single points of failure. The interface would allow for providers to indicate availability in a particular country so that the attestation service can select intelligently.
+
+Attestation Service Configuration:
+```json
+{
+    "providers": [{
+        "type": "twilio",
+        "apiKey": "123ABC"
+    }, {
+        "type": "pool",
+        "apiKey": "123ABC"
+    }],
+}
+```
+
+Attestation Service <-> Pool API Interface
+```
+POST /messages
+
+Request:
+{
+    "dryRun": true, //Allows the service to check if the pool can handle the request before submitting it.
+    "phoneNumber": "+491522931122", // e164-formatted phone number
+    "message": "Your code is XYZ", //the message body to send
+    "address": "0x123", //the account address that is requesting attestation
+}
+
+Responses:
+201 Created
+{
+    "id": "123XYZ",
+    "status": "DISPATCHING"
+}
+
+403 Forbidden
+// When the pool believes it already sent this message
+404 Not Found
+// When the pool could not validate the message
+409 Conflict
+// When the pool believes it cannot service the request, most likely because the pool has no registered verifiers
+410 Gone
+// When the request is no longer valid
+
+GET /messages/#{id}
+
+Responses:
+200 OK
+{
+    "id": "123XYZ",
+    "status": "DISPATCHING"
+}
+
+PATCH /messages#{id}
+Request:
+{
+    "status": "Cancelled" // Indicates if the pool wants to cancel the sending of this message, most likely if the pool is taking too long to indicate successful delivery
+}
+```
+
+Message Statuses:
+- `VALID` - Pool indicates the the message would have been dispatching if dryRun was false
+- `DISPATCHING` - Pool started looking for verifiers to send the text message
+- `ASSIGNED` - Pool has assigned a verifier to send the message
+- `SENT` - The verifier has claimed to have sent the message
+- `CONFIRMED` - The user has completed the attestation
+- `REWARDED` - The verifier has been rewarded
+
+
+**Build a text message provider + app**
+We should effectively revive the pool part of the old verification pool and the verifier app, and adjust it to the new interface to work as a module to the attestation service. Since this service is not expected to run by validators, but by specific entities, building this on top of proprietary tooling is likely ok. We could provide sybil resistance through dedicated APIs such as [https://developer.android.com/training/safetynet/attestation](https://developer.android.com/training/safetynet/attestation) or [https://developer.android.com/training/articles/security-key-attestation](https://developer.android.com/training/articles/security-key-attestation). If we deem these APIs to not be broadly available in our target markets, we could fallback to classic sybil mechnisms like Social Networks, although they are 1. not as sybil resistent and 2. not as permissionless/inclusive.
+
+
+### Alternative Solution
```

Comment:
I'd like to see this whole document following the new template so we can have consistency.